src/gui/layout/qgslayoutlegendwidget.cpp

```diff
@@ -770,8 +772,17 @@ void QgsLayoutLegendWidget::mMoveDownToolButton_clicked()
   }
   else // legend node
   {
-    _moveLegendNode( legendNode->layerNode(), _unfilteredLegendNodeIndex( legendNode ), 1 );
-    mItemTreeView->layerTreeModel()->refreshLayerLegend( legendNode->layerNode() );
+    // get the next index, the one that will have to be above our index,
+    const QModelIndex nextIndex = index.siblingAtRow( index.row() + 1 );
+    if ( nextIndex.isValid() )
+    {
+      QgsLayerTreeModelLegendNode *nextLegendNode = mItemTreeView->index2legendNode( nextIndex );
+      if ( nextLegendNode )
```

Comment:
```suggestion
      QgsLayerTreeModelLegendNode *nextLegendNode = nextIndex.isValid() ? mItemTreeView->index2legendNode( nextIndex ) : nullptr;
      if ( nextLegendNode )
```

Just an esthetic proposal